bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -194,6 +196,29 @@ bool PrePrepareMsg::checkRequests() const {
   Assert(false);
   return false;
 }
+std::string PrePrepareMsg::getMessageClientCorrelationId(int index) {
+  auto it = RequestsIterator(this);
+  int req_num = 0;
+  while (!it.end() && req_num < index) {
+    it.gotoNext();
+    req_num++;
+  }
+  char* requestBody = nullptr;
+  it.getCurrent(requestBody);
+  ClientRequestMsg req((ClientRequestMsgHeader*)requestBody);
+  return req.getCid();
+}
+std::string PrePrepareMsg::getAllRequestsCorrelationIdAsString() {
+  std::stringstream ret;
```

Comment:
Maybe we can use the "+" operator for strings instead of `std::stringstream` in order to avoid copying the output in `std::stringstream::str()` .

Reply:
fixed